### ISSUE TITLE
No "unreachable code" warning when using partial function syntax

### ISSUE THREAD

**Issue by sake92** [lampepfl/dotty#18493]
## Compiler version
3.*

## Minimized code

```Scala
object PartialFunctionNoWarning {

  // nice warning
  "abc" match {
    case "abc" =>
    case "abc" =>
  }
  
  // no warnings
  val pf: PartialFunction[String, Unit] = {
    case "abc" =>
    case "abc" =>
  }

}
```

## Output

"unreachable code" when using `match`.

## Expectation

"unreachable code" shows when using partial function syntax too.  
In scala 2 it does show a warning.

**Comment by jchyb** [comment]
Good news! This appears to be fixed in 3.3.1-RC7 (so will be in 3.3.1) and on main.

**Comment by jchyb** [comment]
Sorry @bishabosha, I think I filled in the labels at the same time as you, causing the mess above
